lib/cocoapods-downloader/http.rb

```diff
@@ -62,6 +65,8 @@ def type_with_url(url)
           :tbz
         elsif path =~ /.(txz|tar\.xz)$/
           :txz
+        elsif path =~ /.dmg$/
```

Comment:
This dot should be escaped (yes, I realise the other regex are also wrong, that's a separate problem).

```diff
- elsif path =~ /.dmg$/
+ elsif path =~ /\.dmg$/
```